src/main/java/com/shapesecurity/shift/validator/Validator.java

```diff
@@ -155,7 +159,21 @@ private boolean isProblematicIfStatement(IfStatement node) {
   public ValidationContext reduceBindingIdentifier(@NotNull BindingIdentifier node) {
     ValidationContext s = super.reduceBindingIdentifier(node);
     if (!checkIsValidIdentifierName(node.name)) {
-      s.addError(new ValidationError(node, "the name field of binding identifier must be a valid identifier name"));
+      if (node.name.equals("*default*") && s.bindingIdentifierNameCanBeDefault) {
+
+      } else {
+        s.addError(new ValidationError(node, "the name field of binding identifier must be a valid identifier name"));
+      }
+    }
+    return s;
+  }
+
+  @NotNull
+  @Override
+  public ValidationContext reduceExportDefault(@NotNull ExportDefault node, @NotNull ValidationContext body) {
+    ValidationContext s = super.reduceExportDefault(node, body);
+    if (node.body instanceof FunctionDeclaration) {
+      s.bindingIdentifierNameCanBeDefault = true;
```

Comment:
By this point, you've already reduced the BindingIdentifier node. This shouldn't work. You're going to need to store errors due to BindingIdentifier nodes being `*default*` in a list in the state object, then clear the list in reduceExportDefault, and merge the list into the errors list at reduceScript and reduceModule.